lib/krb5/principal.c

```diff
@@ -211,6 +211,10 @@ krb5_parse_name_flags(krb5_context context,
     int enterprise = flags & KRB5_PRINCIPAL_PARSE_ENTERPRISE;
     int ignore_realm = flags & KRB5_PRINCIPAL_PARSE_IGNORE_REALM;
     int no_def_realm = flags & KRB5_PRINCIPAL_PARSE_NO_DEF_REALM;
+    int srv_and_inst = 0;
+
+    if (!strncasecmp(name, KRB5_TGS_NAME, 6))
```

Comment:
Later on in this function the comp[] array will be populated with C-strings.   The string that should be compared to KRB5_TGS_NAME is comp[0].